ext/pdo_odbc/odbc_driver.c

```diff
@@ -372,6 +372,26 @@ static int odbc_handle_get_attr(pdo_dbh_t *dbh, zend_long attr, zval *val)
 	return 0;
 }
 
+static zend_result odbc_handle_check_liveness(pdo_dbh_t *dbh)
+{
+	RETCODE ret;
+	UCHAR d_name[32];
+	SQLSMALLINT len;
+	pdo_odbc_db_handle *H = (pdo_odbc_db_handle *)dbh->driver_data;
+
+	/*
+	 *  XXX: Should we be using SQL_ATTR_CONNECTION_DEAD? Procedural ODBC
+	 *  uses this check, but it might be problematic per #20298
```

Comment:
Yes, I think we should check for SQL_ATTR_CONNECTION_DEAD. ext/odbc likely doesn't do this, because `SQLGetConnectAttr ()` is ODBC 3.0 and the ODBC 2.0 `SQLGetConnectOption()` doesn't support this attribute, but ext/odbc is still supposed to work with ODBC 2.0 (something we probably should have changed years ago). ext/pdo_odbc requires ODBC 3.0 anyway.